apps/meteor/app/livechat/server/api/lib/visitors.ts

```diff
@@ -12,7 +13,23 @@ export async function findVisitorInfo({ visitorId }: { visitorId: IVisitor['_id'
 	}
 
 	return {
-		visitor,
+		visitor: await ensureVisitorHasContactId(visitor),
+	};
+}
+
+export async function ensureVisitorHasContactId(visitor: ILivechatVisitor): Promise<ILivechatVisitor> {
+	if (!visitor || visitor.contactId) {
+		return visitor;
```

Comment:
By `!visitor` this will return undefined or null. Is this the expected behavior? 🤔